File: embrace/online-boutique/src/utils/assets.js
import { resolveFrontendPath } from './frontendOrigin'

/** Resolve catalog image paths for the current frontend origin. */
export function assetUrl(path) {
  if (path.startsWith('http')) {
    return path
  }
  if (typeof window === 'undefined') {
    return path
  }
  return resolveFrontendPath(path)
}

File: embrace/online-boutique/flutter/lib/utils/assets.dart
import '../config/app_config.dart';

/// Same as `src/utils/assets.js` — paths under `/static/...` resolved against boutique origin.
String assetUrl(AppConfig config, String path) {
  if (path.startsWith('http')) return path;
  final o = config.frontendOrigin.replaceAll(RegExp(r'/+$'), '');
  final p = path.startsWith('/') ? path : '/$path';
  return '$o$p';
}
